pkg/certificate/providers/types.go

```diff
@@ -80,4 +80,7 @@ type MRCProviderGenerator struct {
 
 	// TODO(#4502): move these to the compat client once we have added these fields to the MRC.
 	KeyBitSize int
+
+	// TODO(#4745): Remove after deprecating the osm.vault.token option.
+	VaultToken string
```

Comment:
Let's call this DefaultVaultToken, since if there's a secret ref on the MRC we should use that.

Reply:
Done

Reply:
I'm a little confused with `DefaultVaultToken` 
Is this a Default OSM sets? That's typically how I read the word "Default" as in, the software has it's own defaults it provides if something is nil or not specified.
If this is the deprecated or user supplied Vault Token can we use a name such as `DeprecatedVaultToken` or something similar. Using the term Deprecated/Deprecation in var names will let other engineers know not to use that variable - and it will make it easy for cleanup later :)